### PR TITLE
Check for support of LoadFont api

### PR DESCRIPTION
**Pull request by nephele-gh**
This is for r1beta4 compatibility

TODO: test this on beta4 once https://review.haiku-os.org/c/haiku/+/6004 is merged

### CODE REVIEW

File: Source/WebCore/platform/graphics/haiku/FontCustomPlatformData.cpp [hunk 1]
```diff
@@ -46,6 +46,10 @@ std::unique_ptr<FontCustomPlatformData> createFontCustomPlatformData(SharedBuffe
 	BFont font;
 
 	status_t result = font.LoadFont(area, buffer.size(), 0);
+
+    if (result == B_NOT_SUPPORTED)
+        return std::unique_ptr<FontCustomPlatformData>(nullptr);
```
[the remark this answers is not in the document]
Comment (pulkomandy) [reply Source/WebCore/platform/graphics/haiku/FontCustomPlatformData.cpp:51]: No, I'm talking about SharedBuffer, the object that is passed as a parameter to this createFontCustomPlatformData. My understanding is that it is already designed to be shared with other processes, and should be implemented as an area. But I don't know if that's actually the case.

File: Source/WebCore/platform/graphics/haiku/FontCustomPlatformData.cpp [hunk 2]
```diff
@@ -46,6 +46,10 @@ std::unique_ptr<FontCustomPlatformData> createFontCustomPlatformData(SharedBuffe
 	BFont font;
 
 	status_t result = font.LoadFont(area, buffer.size(), 0);
+
+    if (result == B_NOT_SUPPORTED)
```
Comment (lonemadmax) [inline Source/WebCore/platform/graphics/haiku/FontCustomPlatformData.cpp:50]: Shouldn't the check be `!= B_OK`? I guess we'll want the same for whatever error we get.

**Comment by pulkomandy** [comment]
If you want to make separate PRs, you have to push the commits to different branches and create a PR for each branch. Life would be simpler if we used Gerrit...

**Comment by pulkomandy** [comment]
Anyway, I'm OK to merge both changes here if you think they are ready.

**Comment by lonemadmax** [comment]
lgtm. Notice though that app_server still needs its fixes in ControlLook, to which I should go back asap, so you may still find cases where controls disappear.

**Pull request merged**